### ISSUE TITLE
Incorrect result when an empty bitset intersects with itself

### ISSUE THREAD

**Issue by notify-bibi** [pascal-lab/Tai-e#121]
### 📝 Overall Description

assert check failed

### 🎯 Expected Behavior

```java
  public void testSizedEmptyBitVectorDontIntersectsItself() {
    BitVector a = new BitVector(1024);
    assertFalse(a.intersects(a));
  }
```

### 🐛 Current Behavior

```java
    @Override
    public boolean intersects(IBitSet set) {
        if (this.isEmpty() || set.isEmpty()) { // added empty check
            return false;
        }
        if (this == set) {
            return true;
        }
        if (!(set instanceof SparseBitSet)) {
            return super.intersects(set);
        }
        SparseBitSet other = (SparseBitSet) set;
        return iterateBlocks(this, other, new IntersectsAction(this));
    }
```

### ⚙️ Tai-e Arguments

<details>
  <summary>🔍 Click here to see Tai-e Options</summary>
  

  ```yaml
  {{The content of 'output/options.yml' file}}
  ```
</details>
<details>
  <summary>🔍 Click here to see Tai-e Analysis Plan</summary>
  

  ```yaml
  {{The content of 'output/tai-e-plan.yml' file}}
  ```
</details>

### 📜 Tai-e Log

<details>
  <summary>🔍 Click here to see Tai-e Log</summary>
  

  ```
  {{The content of 'output/tai-e.log' file}}
  ```
</details>

**Comment by notify-bibi** [comment]
also check the `public boolean SparseBitSet#contains(IBitSet set)`

**Comment by zhangt2333** [comment]
Thanks for writing this issue in Chinese, it makes it very easy for a few of us developers to read. However, Tai-e is a project with an international focus, and we aim to keep our community discussions in as widely accessible a language as possible, which is English. This helps us share our open-source progress with interested individuals from all over the world. Kindly consider using English for communication in the future. We appreciate your understanding. Thank you.

---

It seems that there is no class named BitVector in Tai-e. Please verify that you're submitting the issue to the correct project.

**Comment by notify-bibi** [comment]
The intersects method in src/main/java/pascal/taie/util/collection/SparseBitSet.java

**Comment by zhangt2333** [comment]
Fixed by 7411dda919ba903b1c1e21fd66ee83a62a098e07.Thanks. ♥️
